src/aiy/assistant/device_helpers.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Device registration helpers for the Google Assistant API."""
+
+import json
+import os
+import random
+
+import google.auth.transport
+
+import aiy.assistant.auth_helpers
+
+
+_DEVICE_MODEL = "voice-kit"
+_DEVICE_MANUFACTURER = "AIY Projects"
+_DEVICE_NAME = "Voice Kit"
+_DEVICE_TYPE = "action.devices.types.LIGHT"
+
+_DEVICE_ID_FILE = os.path.join(
+        aiy.assistant.auth_helpers._VR_CACHE_DIR, 'device_id.json')
+
+
+def _get_project_id():
+    with open(aiy.assistant.auth_helpers._ASSISTANT_CREDENTIALS_FILE) as f:
+        client_secrets_data = json.load(f)
+        return client_secrets_data["installed"]["project_id"]
+
+
+def _get_api_url(*args):
+    return "/".join(
+            ("https://embeddedassistant.googleapis.com/v1alpha2/projects",) + args)
+
+def _load_ids(id_path):
+    with open(id_path, 'r') as f:
+        id_data = json.load(f)
+    return id_data["model_id"], id_data["device_id"]
+
+
+def _save_ids(id_path, model_id, device_id):
+    if not os.path.exists(os.path.dirname(id_path)):
+        os.makedirs(os.path.dirname(id_path))
+
+    id_data = {
+            "model_id": model_id,
+            "device_id": device_id,
+    }
+    with open(id_path, 'w') as f:
+        json.dump(id_data, f)
+
+
+def _get_model_id(credentials, session, project_id):
+    model_id = "%s-%s" % (project_id, _DEVICE_MODEL)
```

Comment:
This was smart to concatenate project_id in there! Since the model namespace is global (not per project) this ensure there will be no conflict :)

Reply:
Not especially smart, I just followed the suggestion in the docs :)